### ISSUE TITLE
How to add support for more custom models in pre-training

### ISSUE THREAD

**Issue by zero0kiriyu** [keyu-tian/SparK#17]
This project is very cool, but it would be nice if you provided a simple tutorial to support more model architectures  :)

**Comment by keyu-tian** [comment]
@zero0kiriyu Thank you for this advice. For how to add a new model u could check https://github.com/keyu-tian/SparK/tree/main/pretrain#some-details-how-we-mask-images-and-how-to-set-the-patch-size, and we're gonna refactor the code (make it more easier to use a custom model) and add a tutorial maybe by next Tuesday. Would comment here again when we finish this.

**Comment by zero0kiriyu** [comment]
It would be great! Also, when I tried to modify the code, I noticed that the number of channels in each layer must be reduced by half, but not all models follow this rule. For example, some lightweight networks used in my projects have channel numbers like 24, 48, 96, 128. Perhaps you can provide more flexibility to support more custom models.

**Comment by keyu-tian** [comment]
@zero0kiriyu You can kindly read https://github.com/keyu-tian/SparK/tree/main/pretrain#tutorial-for-customizing-your-own-cnn-model for how to customize your cnn model, hope you enjoy it!
